demos/sample-javascript/src/main.js

```diff
@@ -8,27 +9,54 @@ const hostUrlEl = document.getElementById("hostUrl");
 const whoAmIResponseEl = document.getElementById("whoamiResponse");
 const canisterIdEl = document.getElementById("canisterId");
 const principalEl = document.getElementById("principal");
+const delegationEl = document.getElementById("delegation");
+const expirationEl = document.getElementById("expiration");
 const iiUrlEl = document.getElementById("iiUrl");
+const maxTimeToLiveEl = document.getElementById("maxTimeToLive");
 
 let authClient;
 
 const init = async () => {
   authClient = await AuthClient.create();
-  principalEl.innerText = await authClient.getIdentity().getPrincipal();
 
-  // Redirect to the identity provider
+  const updateView = () => {
+    const identity = authClient.getIdentity();
+
+    principalEl.innerText = identity.getPrincipal();
+    if (identity instanceof DelegationIdentity) {
+      delegationEl.innerText = JSON.stringify(identity.getDelegation().toJSON());
+
+      const nextExpiration =
+        identity.getDelegation().delegations
+         .map(d => d.delegation.expiration)
+         .reduce((current, next) => next < current ? next : current);
```

Comment:
```suggestion
        Math.min(...identity.getDelegation().delegations.map(d => d.delegation.expiration));
```

Might be a little easier for the non FP folks

Reply:
I actually copied that from the `dfinity_wallet` code :-)